python_scripts/02_numerical_pipeline_introduction.py

```diff
@@ -172,7 +172,10 @@
 (target == target_predicted).mean()
 
 # %% [markdown]
-# But, can this evaluation be trusted, or is it too good to be true?
+# This result means that the model makes a correct _prediction_ for
+# approximately 82 samples out of 100. But, can a model _predict_ something
+# that it already saw? In other words, can this evaluation be trusted, or is it
```

Comment:
I think I know what you are trying to say (that we are measuring the accuracy on the training data and that it is kind of cheating) but I find the wording super confusing ... 

In particular: "can a model predict something that it already saw?" I would answer "yes why not, sorry is this a trick question?"

I think you probably mean "can this really be called prediction when we are learning and predicting from the same data" but I can't find a good wording that convinces me.

I kind of think the next section in train-test data split explains this kind of thing already so I would stay short maybe something like this:

Note that here we used the same data to learn and evaluate our model, so can this evaluation be trusted, or is it too good to be true?